community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/NodeCorrectlyIndexedCheck.java

```diff
@@ -114,6 +117,24 @@ private void verifyNodeCorrectlyIndexedUniquely(
         }
     }
 
+    private PrimitiveLongIterator filterIfMultipleValuesFound( int propertyKeyId, Object propertyValue,
+            PrimitiveLongIterator indexedNodeIds )
+    {
+        if ( ((PrimitiveLongPeekingIterator) indexedNodeIds).hasMultipleValues() )
```

Comment:
But what I found is that the type cast here is not good, so I will make the cast inside this method.